### PR TITLE
Bugfixes

### LINKED ISSUE #62: Eval error with Rc
When I run the following code I get a panic.

## Code:
```rust
use std::cell::RefCell;
use std::rc::Rc;

fn rc_refcell() -> i32 {
     let r = Rc::new(RefCell::new(42));
     *r.borrow_mut() += 10;
     let x = *r.borrow();
     x
}

fn main() {
    rc_refcell();
}
```

## Backtrace:
```
    Finished debug [unoptimized + debuginfo] target(s) in 0.1 secs
     Running `target/debug/miri example_code.rs`
thread 'main' panicked at 'assertion failed: `(left == right)` (left: `1`, right: `0`)', src/interpreter/mod.rs:535
stack backtrace:
   1: 0xb672ec11 - std::sys::backtrace::tracing::imp::write::h4b09e6e8c01db097
   2: 0xb673edaa - std::panicking::default_hook::{{closure}}::h1d3243f546573ff4
   3: 0xb673b949 - std::panicking::default_hook::h96c288d728df3ebf
   4: 0xb673c053 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
   5: 0xb673bedd - std::panicking::begin_panic::hfbeda5aad583dc32
   6: 0xb673bdfe - std::panicking::begin_panic_fmt::h4fe9fb9d5109c4bf
   7: 0xb75c6794 - miri::interpreter::EvalContext::eval_rvalue_into_lvalue::h1e311a0af7ed05c7
                at /home/bjorn/Documenten/miri/src/interpreter/mod.rs:535
   8: 0xb759e79d - miri::interpreter::step::<impl miri::interpreter::EvalContext<'a, 'tcx>>::statement::he6295ec356cfef01
                at /home/bjorn/Documenten/miri/src/interpreter/step.rs:80
   9: 0xb759db8b - miri::interpreter::step::<impl miri::interpreter::EvalContext<'a, 'tcx>>::step::ha1e6c1f76ff81898
                at /home/bjorn/Documenten/miri/src/interpreter/step.rs:47
  10: 0xb75d8cfd - miri::interpreter::eval_main::h2893a72c73e2909c
                at /home/bjorn/Documenten/miri/src/interpreter/mod.rs:1171
  11: 0xb73d2369 - <miri::MiriCompilerCalls as rustc_driver::CompilerCalls<'a>>::build_controller::{{closure}}::h168a492b582a289e
                at /home/bjorn/Documenten/miri/src/bin/miri.rs:78
  12: 0xb7292b0f - rustc_driver::driver::compile_input::{{closure}}::hdf708031c67974d4
  13: 0xb728287b - rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}::hcac3a2328cd9259a
  14: 0xb71c5f3c - rustc::ty::context::TyCtxt::create_and_enter::h391a9cd3e27a0125
  15: 0xb7243411 - rustc_driver::driver::compile_input::hc0edbed7edb3eb18
  16: 0xb72693ff - rustc_driver::run_compiler::h22d678d32fb7c300
  17: 0xb73d09ce - miri::main::he47b4050ae2aa0ba
                at /home/bjorn/Documenten/miri/src/bin/miri.rs:141
  18: 0xb673bca5 - std::panicking::try::do_call::h8a9d45772be711b8
  19: 0xb67468e7 - __rust_maybe_catch_panic
  20: 0xb673aee6 - std::rt::lang_start::haaae1186de9de8cb
  21: 0xb73d253a - main
  22: 0xb6495af2 - __libc_start_main
  23: 0xb736c140 - <unknown>
error: Process didn't exit successfully: `target/debug/miri example_code.rs` (exit code: 101)
```

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #62 and #63 and does a rustup to the unreleased nightly

### CODE REVIEW

File: src/interpreter/mod.rs [hunk 1]
```diff
@@ -31,6 +32,10 @@ pub struct EvalContext<'a, 'tcx: 'a> {
     /// The results of the type checker, from rustc.
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
 
+    /// The Session, from rustc.
+    /// Used to extract info from other crates
+    session: &'a Session,
```
Comment (eddyb) [inline src/interpreter/mod.rs]: This is `tcx.sess`.

**Comment by eddyb** [comment]
FWIW `get_vtable_methods` is already in `rustc::traits`.

File: src/interpreter/mod.rs (hunk 1 above)
Comment (oli-obk) [inline src/interpreter/mod.rs]: darn it ^^ I thought it was somewhere there, but couldn't find it. Rustdocs are overwhelmed with TyCtxt

**Comment by oli-obk** [comment]
> FWIW get_vtable_methods is already in rustc::traits.

it is... but I can't use it yet, because of the reification issues

**Comment by solson** [comment]
Travis must have failed due to not having the latest nightly. Builds locally.

**Comment by bjorn3** [comment]
Thanks @oli-obk!

**Pull request merged**